doc/compiler/il/MainlineAndOutOfLineCode.md

```diff
@@ -0,0 +1,30 @@
+<!--
+Copyright (c) 2021, 2021 IBM Corp. and others
+
+This program and the accompanying materials are made available under
+the terms of the Eclipse Public License 2.0 which accompanies this
+distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+or the Apache License, Version 2.0 which accompanies this distribution and
+is available at https://www.apache.org/licenses/LICENSE-2.0.
+
+This Source Code may also be made available under the following
+Secondary Licenses when the conditions for such availability set
+forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+General Public License, version 2 with the GNU Classpath
+Exception [1] and GNU General Public License, version 2 with the
+OpenJDK Assembly Exception [2].
+
+[1] https://www.gnu.org/software/classpath/license.html
+[2] http://openjdk.java.net/legal/assembly-exception.html
+
+SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+-->
+
+# Mainline and out of line code
```

Comment:
This is a good start, although it needs to be broken down. There are really two types of outlined code:

1. Global (i.e. blocks)
2. Local (Out Of Line (OOL) code sections)

This document should in my opinion have the above two sections and describe the difference. For grabl outlined code really means cold blocks and talk about how they originate and which opts (Cold Block Marking) are responsible for shuffling the blocks to ensure the "hot" code is kept close together.

For local OOL I often reference this giant comment we wrote on the topic:
https://github.com/eclipse/omr/blob/7c7d075d097816da7693d7eb0c318b6d4d903b8f/compiler/codegen/OutOfLineCodeSection.hpp#L36-L106

Reply:
This doesn't have to be done by you by the way, nor does it have to be done now. Just putting some thoughts on how we can improve this documentation.

Thanks for making this!

Reply:
Good suggestion! I think I could dedicate this PR for `GlRegDeps.md`. I'll work on `MainlineAndOutOfLineCode.md`  with the material you've mentioned above and put it in a future PR when it's ready.